### PR TITLE
make chunk_extra referenced by shard_uid instead of shard_id

### PR DESCRIPTION
**Pull request by mzhangmzz**
To support re-sharding, at the epoch where splitting states happens, the validators need to keep chunk extra for the shards for this epoch and the shards in the coming epoch. This PR changes chunk_extra to be stored and referenced by `shard_uid` instead of `shard_id`

### CODE REVIEW

File: chain/chain/src/store.rs [hunk 1]
```diff
@@ -2077,7 +2077,11 @@ impl<'a> ChainStoreUpdate<'a> {
             self.gc_col(ColIncomingReceipts, &block_shard_id);
             self.gc_col(ColChunkPerHeightShard, &block_shard_id);
             self.gc_col(ColNextBlockWithNewChunk, &block_shard_id);
-            self.gc_col(ColChunkExtra, &block_shard_id);
+
+            // TODO: use the real shard version
```
Comment (bowenwang1996) [inline chain/chain/src/store.rs:2081]: is there an issue for this?

**Review by bowenwang1996 — APPROVED** [review]

File: chain/chain/src/store.rs (hunk 1 above)
Comment (mzhangmzz) [reply chain/chain/src/store.rs:2081]: What do you mean?

I created an issue for this TODO https://github.com/near/nearcore/issues/4710. I will implement that in a later PR.

One issue is that state validator will fail if these columns are not garbage collected correctly. One way to hack that is to garbage collect both `shard_version` 0 and 1. The most correct way is pass the correct shard layout version to the gc function.

Comment (bowenwang1996) [reply chain/chain/src/store.rs:2081]: Could we add the issue number here?

**Review by chefsale — APPROVED** [review]

**Review by ailisp — APPROVED** [review]
Looks good to me!

**Comment by bowenwang1996** [comment]
@nikurt @Longarithm please review

**Review by nikurt — APPROVED** [review]

**Review by Longarithm — APPROVED** [review]

**Pull request merged**